### PR TITLE
layers: Add WSI support to parameter_validation

### PR DESCRIPTION
**Pull request by mark-lunarg**
Enabled Parameter Validation of WSI functions.  The swapchain layer PR #834 is dependent upon this change.

I've run Windows and Linux here, but it'd be great if you could run it through your farm, please.

### CODE REVIEW

File: layers/parameter_validation.cpp [hunk 1]
```diff
@@ -71,11 +71,14 @@ struct layer_data {
...
 
+    bool wsi_enabled;
+
     layer_data()
         : report_data(nullptr), num_tmp_callbacks(0), tmp_dbg_create_infos(nullptr), tmp_callbacks(nullptr), device_limits{},
-          physical_device_features{}, physical_device{} {};
+          physical_device_features{}, physical_device{}, wsi_enabled(false) {};
 };
 
+static std::unordered_map<void *, struct instance_extension_enables> instanceExtMap;
```
Comment (tobine) [inline layers/parameter_validation.cpp]: No camelCase for new var names please

File: layers/parameter_validation.cpp [hunk 2]
```diff
@@ -1583,6 +1586,59 @@ void validateDeviceCreateInfo(VkPhysicalDevice physicalDevice, const VkDeviceCre
     }
 }
 
+static void CheckInstanceRegisterExtensions(const VkInstanceCreateInfo *pCreateInfo, VkInstance instance) {
+    VkLayerInstanceDispatchTable *pDisp = get_dispatch_table(pc_instance_table_map, instance);
```
Comment (tobine) [inline layers/parameter_validation.cpp]: Kill da camel... case

File: layers/parameter_validation.cpp [hunk 3]
```diff
@@ -1583,6 +1586,59 @@ void validateDeviceCreateInfo(VkPhysicalDevice physicalDevice, const VkDeviceCre
...
+#ifdef VK_USE_PLATFORM_WIN32_KHR
+        if (strcmp(pCreateInfo->ppEnabledExtensionNames[i], VK_KHR_WIN32_SURFACE_EXTENSION_NAME) == 0) {
+            instanceExtMap[pDisp].win32_enabled = true;
+        }
+#endif
+    }
+}
+
+static void CheckDeviceRegisterExtensions(const VkDeviceCreateInfo *pCreateInfo, VkDevice device) {
+    layer_data *device_data = get_my_data_ptr(get_dispatch_key(device), layer_data_map);
```
Comment (tobine) [inline layers/parameter_validation.cpp]: This is the most beautifully named variable that I've seen in my entire life

File: layers/parameter_validation.cpp [hunk 4]
```diff
@@ -4614,12 +4671,374 @@ VKAPI_ATTR VkResult VKAPI_CALL EnumerateDeviceExtensionProperties(VkPhysicalDevi
         ->EnumerateDeviceExtensionProperties(physicalDevice, NULL, pCount, pProperties);
 }
 
+
+// WSI Extension Functions
+
+VKAPI_ATTR VkResult VKAPI_CALL CreateSwapchainKHR(VkDevice device, const VkSwapchainCreateInfoKHR *pCreateInfo,
+                                                    const VkAllocationCallbacks *pAllocator, VkSwapchainKHR *pSwapchain) {
+    VkResult result = VK_ERROR_VALIDATION_FAILED_EXT;
+    bool skipCall = false;
```
Comment (tobine) [inline layers/parameter_validation.cpp]: Please bulk replace skipCall w/ "skip_call" or "skip" as @chrisforbes boldly used recently.

**Comment by tobine** [comment]
Clean on farm. Just a few var naming fix-ups and LGTM

**Comment by mark-lunarg** [comment]
Added all of Tobin's fixups and also ran clang-format on the thing, then pushed it to Master.

**Pull request merged**